lib/takeaway.rb

```diff
@@ -0,0 +1,32 @@
+require 'order'
+require 'sms'
+
+class Takeaway
+  attr_reader :order, :sms, :menu
+
+  def initialize(order = Order.new(menu), sms = nil , config = {})
+   
+    @order = order
+    @sms = sms || SMS.new(config)
+    # this is order object
+    # @total = 0
+  end 
+
+  def check_menu(all_dishes = Menu.new)
```

Comment:
Where is the Menu.new coming from? It has not been injected and the menu file has not been required in this file?